pkg/apis/configuration/validation/validation.go

```diff
@@ -707,6 +737,124 @@ func validateVariable(nVar string, validVars map[string]bool, fieldPath *field.P
 		msg := fmt.Sprintf("'%v' contains an invalid NGINX variable. Accepted variables are: %v", nVar, mapToPrettyString(validVars))
 		allErrs = append(allErrs, field.Invalid(fieldPath, nVar, msg))
 	}
+	return allErrs
+}
+
+func isValidSpecialVariableHeader(header string) []string {
+	errMsgs := validation.IsHTTPHeaderName(strings.Replace(header, "_", "-", -1))
```

Comment:
modifying the input so that we can use an existing function for validation adds complexity - it makes the code harder to understand. 
I suggest adding a comment about why we need replace `_` with `-`. For example,  
```
# underscores in $http_ variable represent '-'.  
```

Reply:
I don't agree, but thanks for the suggestion